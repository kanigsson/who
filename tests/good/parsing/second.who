type bool
logic true : bool
logic false : bool
logic == ['a||] : 'a -> 'a -> bool
let f ['a||] (x y : 'a) = 
  {} 
  x == y 
  {}

let f ['a||] (x : 'a) = {} true {}

let g = f true
let h = f 5
