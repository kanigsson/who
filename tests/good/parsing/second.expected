let f ['a]   = (λ(x:'a) -> (λ(y:'a) -> ((beq_z x) y))) in
let f ['a]   = (λ(x:'a) -> true) in
let g   = (f true) in let h   = (f 5) in ()
